### PR TITLE
chore(gatsby): Migrate utils/tracer/zipkin-local to TypeScript

### PR DESCRIPTION
**Pull request by kawamataryo**
## Description

Convert `utils/tracer/zipkin-local` to typescript.

## Related Issues

Related to #21995

### CODE REVIEW

**Comment by kawamataryo** [comment]
It doesn't seem to need to be TypeScript?
It used by Local zipkin with docker. https://github.com/gatsbyjs/gatsby/blob/c9b0fa3fceab7ec3dd4b4d1be762fd8c470b5cd9/docs/docs/performance-tracing.md#local-zipkin-with-docker

**Comment by blainekasten** [comment]
@kawamataryo we can still make this a TS file! Let's do this 💜

File: packages/gatsby/package.json [hunk 1]
```diff
@@ -145,7 +145,10 @@
     "webpack-merge": "^4.2.2",
     "webpack-stats-plugin": "^0.3.1",
     "xstate": "^4.8.0",
-    "yaml-loader": "^0.5.0"
+    "yaml-loader": "^0.5.0",
+    "zipkin": "^0.19.2",
+    "zipkin-transport-http": "^0.19.2",
+    "zipkin-javascript-opentracing": "^2.1.0"
```
Comment (blainekasten) [inline packages/gatsby/package.json]: I think we need to move these to dev deps

Comment (kawamataryo) [reply packages/gatsby/package.json]: You’re right. I moved  Those  to devs.

https://github.com/gatsbyjs/gatsby/pull/22372/commits/7d68a33a8f1c4eb84f7ca975ece1944239829b18

**Pull request merged**